### ISSUE TITLE
Default Binding Selection

### ISSUE THREAD

**Issue by eitanlees** [altair-viz/altair#1121]
I know binding documentation is on the way but I have question nonetheless.  (I will include a "copy/paste"-able example at the end for convenience.) 

### Background
```python
import numpy as np
import pandas as pd
import altair as alt
from vega_datasets import data
```
Let's say I am interested in analysing some stock information. I am interested in a single stock as well as a few indicator variables A B and C. 

```python
stocks = data.stocks()
apple = stocks.loc[stocks['symbol']=='AAPL',:].copy()

np.random.seed(0)
apple['A'] = np.random.normal(size=len(apple)).cumsum() + 50
apple['B'] = np.random.normal(size=len(apple)).cumsum()
apple['C'] = np.random.normal(size=len(apple)).cumsum()
apple.head()
```

| symbol | date | price | A | B | C|
| ------- | ----- | ----- | -- | -- | -- |
AAPL | 2000-01-01 | 25.94 | 51.764052 | 1.326386 | 1.152332
AAPL | 2000-02-01 | 28.66 | 52.164210 | 0.631818 | 2.231950
AAPL | 2000-03-01 | 33.95 | 53.142948 | 0.482183 | 1.418586
AAPL | 2000-04-01 | 31.01 | 55.383841 | 0.047030 | -0.047838
AAPL | 2000-05-01 | 21.00 | 57.251399 | 1.896294 | 0.473226

Before heading to altair I want to `melt()` the data so that it is in [long-form](https://altair-viz.github.io/user_guide/data.html#long-form-vs-wide-form-data). 
```python
apple = apple.melt(['date', 'price', 'symbol'])
apple.head()
```
| date | price | symbol | variable | value|
| ---- | ------ | ------- | -------- | ---- |
2000-01-01 | 25.94 | AAPL | A | 51.764052
2000-02-01 | 28.66 | AAPL | A | 52.164210
2000-03-01 | 33.95 | AAPL | A | 53.142948
2000-04-01 | 31.01 | AAPL | A | 55.383841
2000-05-01 | 21.00 | AAPL | A | 57.251399

It's important to note that the indicator variables are different orders of magnitude. 
```python
alt.Chart(apple).mark_line().encode(y='value', x='date', color='variable')
```
![indicators](https://user-images.githubusercontent.com/2041969/44731805-f91a8c00-aab1-11e8-93e1-815b4aba75e4.png)

### Problem Statement
I would like to create a plot with a time series of the price information and a heatmap of the indicator information. I want the time series to include a selection for the date range, and I want the heatmap to include a drop down menu for which indicator variable to display.

Here was my idea 

```python
date_selection = alt.selection_interval(encodings=['x'])

toggle = alt.binding_select(options=['A', 'B', 'C'])
variable_selection = alt.selection_single(fields=['variable'], bind=toggle, name='Select_')

heat_map = alt.Chart().mark_rect().encode(
    x = 'year(date):O', 
    y = 'month(date):O', 
    color = alt.condition(date_selection, 'value:Q', alt.ColorValue('lightgray'))
).add_selection(
    variable_selection
).transform_filter(
    variable_selection
)

time_series = alt.Chart().mark_line().encode(
    x='date:T', 
    y='price:Q'
).add_selection(date_selection)

alt.vconcat(time_series, heat_map, data=apple).resolve_scale('independent')
```
![screen shot 2018-08-28 at 11 10 32 am](https://user-images.githubusercontent.com/2041969/44732498-6c70cd80-aab3-11e8-8e50-9ff58ebcf20b.png)

Here is where the weird behavior is happening. Clicking between the heatmap and the time series seems to change what is displayed. The scale appears to be for the entire dataset. I can't seem to figure out what is going wrong. 

### TLDR

- Why does clicking the heatmap change what is being plotted?
- Can you set a default value for the drop down binding?
- Is it ok to mix selection intervales with single selections?

As the binding documentation comes together I am sure things will start to make more sense. 

Thanks for your time and help!

### Full Example Code
```python
import numpy as np
import pandas as pd
import altair as alt
from vega_datasets import data

stocks = data.stocks()
apple = stocks.loc[stocks['symbol']=='AAPL',:].copy()

np.random.seed(0)
apple['A'] = np.random.normal(size=len(apple)).cumsum() + 50
apple['B'] = np.random.normal(size=len(apple)).cumsum()
apple['C'] = np.random.normal(size=len(apple)).cumsum()

apple = apple.melt(['date', 'price', 'symbol'])

date_selection = alt.selection_interval(encodings=['x'])

toggle = alt.binding_select(options=['A', 'B', 'C'])
variable_selection = alt.selection_single(fields=['variable'], bind=toggle, name='Select_')

heat_map = alt.Chart().mark_rect().encode(
    x = 'year(date):O', 
    y = 'month(date):O', 
    color = alt.condition(date_selection, 'value:Q', alt.ColorValue('lightgray'))
).add_selection(
    variable_selection
).transform_filter(
    variable_selection
)

time_series = alt.Chart().mark_line().encode(
    x='date:T', 
    y='price:Q'
).add_selection(date_selection)

alt.vconcat(time_series, heat_map, data=apple).resolve_scale('independent')
```

**Comment by jakevdp** [comment]
> Why does clicking the heatmap change what is being plotted?

Because you added the single selection to the heatmap – this makes it so that clicking a mark on the heatmap affects the selection. Think of the drop-down widget as an extra way to access this selection.

I hadn't actually appreciated this before... the same thing happens in our binding example if you click on the bars: https://altair-viz.github.io/gallery/us_population_pyramid_over_time.html I wonder if there is a way to add a selection to a chart *without* the default behavior of linking the selection to the marks themselves? (@arvind might know)

> Can you set a default value for the drop down binding?

No, not at this time – see https://github.com/vega/vega-lite/pull/4139

> Is it ok to mix selection intervals with single selections?

Yes, mixing selections is fine. I don't think that's causing any issues here.

**Comment by hassenmorad** [comment]
@eitanlees quick question... I entered your code in my notebook and found that the 'date' column values were displayed as M-D-Y. How can I get it to display just the year like in your plot?

**Comment by jakevdp** [comment]
I think you may be using an older version of the vega extensions. Newer version show the year only by default.

If you want to directly control the format of the tick labels, you can use format codes in the ``axis`` property. For example, replace
```py
x='date:T'
```
with
```py
x=alt.X('date:T', axis=alt.Axis(format="%Y"))
```

**Comment by hassenmorad** [comment]
@jakevdp Thanks, that did the trick!

**Comment by jakevdp** [comment]
Initialization of selections is now supported in Altair v3. For example: https://altair-viz.github.io/gallery/us_population_over_time.html
